std/algorithm/searching.d

```diff
@@ -2361,19 +2361,49 @@ If $(D haystack) is a random-access range, all three components of the
 tuple have the same type as $(D haystack). Otherwise, $(D haystack)
 must be a forward range and the type of $(D result[0]) and $(D
 result[1]) is the same as $(XREF range,takeExactly).
+
+Returns:
+A sub-type of Tuple!() with opCast defined for bool.  This opCast returns $(D
+true) when the separating $(D needle) was found (!result[1].empty) and $(D
+false) otherwise.  This enables convenient usage in pattern matching shown in
+the following example
+
+Example:
+---
+if (const split = haystack.findSplit(needle))
+{
+     doSomethingWithSplit(split);
+}
+---
  */
 auto findSplit(alias pred = "a == b", R1, R2)(R1 haystack, R2 needle)
 if (isForwardRange!R1 && isForwardRange!R2)
 {
+    static struct Result(R1, R2) if (isForwardRange!R1 &&
+                                     isForwardRange!R2)
+    {
+        this(R1 pre, R1 separator, R2 post)
```

Comment:
No need for these to be templates.

No need for these constructors, just use `Result(tuple(....))`.